src/core/PRINCIPLES.md

```diff
@@ -0,0 +1,47 @@
+## Common dictionary
+Plugin - a piece of software expressing specific subject of the business unit within the code.
+
+Platform - a set of plugins required to run all other Kibana plugins.
+
+## New platform principles
+### Explicit business domains separation
+The plugins code should be structured on the top level to reflect business units.
+```js
+// GOOD
+src/plugins
+ - apm
+  - server
+  - public
+ - search
+ ..
+// BAD
+src/plugins
+ - server
+  - apm
+  - search
+```
+### Explicit dependencies
+Each plugin should declare dependencies on the other plugins explicitly. Plugins cannot have circular dependencies. Plugins shouldn't have hidden dependencies.
+```json
+"requiredPlugins": ["b"],
+"optionalPlugins": ["c"],
+```
+### Explicit API declaration
+Each plugins defines API & extension points explicitly. Any other API that not declared explicitly as public should be considered private. HTTP endpoints belonging to other plugins are considered private. Plugins should expose JavaScript interfaces on top of these HTTP endpoints for other plugins to consume.
```

Comment:
Is there a difference between "API" and "extension point"?
What about saying "should expose JavasScript client" instead of interface?
```suggestion
Each plugin has to define an explicit API. Any other API's that has not been declared explicitly as public should be considered private. HTTP endpoints belonging to other plugins are considered private. Plugins should expose a JavaScript client on top of these HTTP endpoints for other plugins to consume.
```